quinn/src/streams.rs

```diff
@@ -330,28 +330,22 @@ impl Future for ReadToEnd {
     type Error = ReadError;
```

Comment:
The fact that we have the `TooLong` error here suggests to me that maybe a generic `ReadError` isn't quite right for `ReadToEnd` and we should maybe have a wrapper with `Read` and `TooLong` variants?

Reply:
I agree that the error case does not belong in the generic `ReadError`; `ReadToEnd` is intended to be a helper that could just as well well exist outside of the crate entirely.

Reply:
Fixed that up.